### PR TITLE
Allow script to be sourced

### PR DESCRIPTION
**Pull request by andybouts**
Resolve Jira IAC-248

### CODE REVIEW

**Comment by andybouts** [comment]
This was cleaned up per request

File: files/tools/terraform_env_variable_helper.sh [hunk 1]
```diff
@@ -1,16 +1,18 @@
 #!/usr/bin/env bash
+# `set -e` is used below to ensure that all of the assignments in the script are made / that it runs in entirety
 set -e
 
+# usage: run the export in either line 8 or 9 before this script or edit lines 8 or 9 in this script before running
 # usage: source <path-to-file>TerraformEnvVariableAssignment.sh
 # note: select one of the following options for your Service Principal
 export YourSP="<yourSP>"
```
Comment (thpang) [inline files/tools/terraform_env_variable_helper.sh:8]: This item should follow stand ENV variable formatting. All caps. The value should be: YOURSP. Also this line should be removed. The ENV should be required by the script of it fails.

File: files/tools/terraform_env_variable_helper.sh [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
 #export YourSP=$USER
 
-echo -e "You must have an active az cli login 'az login' before this script will work"
-echo -e "We will use the Service Principal for >>>>> $YourSP <<<<<"
-echo -e "Otherwise edit the export YourSP earlier in this script "
-echo -e "\nPausing for 5 seconds; use ctrl-c to exit and login"
-sleep 5s
+echo -e "Usage: You must have an active az cli login 'az login' before this script will work"
+echo -e "Usage: Run the export in either line 8 or 9 before this script or edit lines 8 or 9 in this script before running"
+echo -e "Usage: The variable \$YourSP needs to be set and currently has a value of Service Principal: >>>>> $YourSP <<<<<"
```
Comment (thpang) [inline files/tools/terraform_env_variable_helper.sh:13]: Again $YourSP -> $YOURSP

File: files/tools/terraform_env_variable_helper.sh [hunk 3]
```diff
@@ -42,7 +44,13 @@ export TF_VAR_client_secret
...
+# `set +e` reverses the `set -e`
+set +e
+
 echo -e "\nAll done"
 echo -e "\nIf the wrong subscription was selected, then suggest:"
 echo -e "   \$az account list --output table"
 echo -e "   \$az account set --subscription <YourSubscription>"
+
+echo -e "\nIf the azurerm module is returning an http/401 error, then you might want to reset your SP credential:"
+echo -e "   \$az ad sp credential reset --name $YourSP"
```
Comment (thpang) [inline files/tools/terraform_env_variable_helper.sh:56]: $YourSP -> $YOURSP

**Review by thpang — COMMENTED** [review]
The YourSP needs to be a true ENV variable. Comments below.

**Comment by andybouts** [comment]
Closing this since I could not unstage a commit on gitignore

**Pull request closed without merging**